lib/run/frame.ts

```diff
@@ -17,83 +18,182 @@ export interface FrameOptions<T> {
 }
 
 export function createFrame<T>(options: FrameOptions<T>): Frame<T> {
-  let { operation, parent } = options;
-  let children = new Set<Frame>();
-  let block = createBlock(operation);
-  let context = Object.create(parent?.context ?? {});
-  let results = createEventStream<void, Result<void>>();
+  return evaluate<Frame<T>>(function* () {
+    let { operation, parent } = options;
+    let children = new Set<Frame>();
+    let context = Object.create(parent?.context ?? {});
+    let thunks: IteratorResult<Thunk, Result<T>>[] = [{
+      done: false,
+      value: $next(void 0),
+    }];
+
+    let crash: Error | undefined = void 0;
+
+    let interrupt = () => {};
+
+    let abort = (reason?: Error) => {
+      if (!frame.aborted) {
+        frame.aborted = true;
```

Comment:
Here you are referencing a `frame` before it has been created -- this is confusing to reason about.  Further, this function is used inside `frame.crash` and `frame.destroy`.  This is another example of a circular reference that makes me nervous.

Reply:
All of the circular references were removed.